lib/browser.ts

```diff
@@ -672,17 +673,14 @@ export class ProtractorBrowser extends Webdriver {
 
     if (this.ignoreSynchronization) {
       this.driver.get(destination);
-      return this.driver.controlFlow().execute(() => {
-        return this.plugins_.onPageLoad();
-      });
+      return this.driver.controlFlow().execute(() => this.plugins_.onPageLoad()).then(() => {});
```

Comment:
Okay, I found that the array is meaningful for `waitForCondition`... It's used in `browser.ts`:
```ts
return this.plugins_.waitForCondition().then((results: boolean[]) => {
  return results.reduce((x, y) => {
    return x && y;
  }, true);
});
```
Which means `.then(() => {})` can't be moved to `pluginFunFactory`.